x-pack/legacy/plugins/siem/server/lib/alerts/elasticsearch_adapter.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get, getOr } from 'lodash/fp';
+
+import { AlertsOverTimeData, MatrixOverTimeHistogramData } from '../../graphql/types';
+
+import { inspectStringifyObject } from '../../utils/build_query';
+
+import { FrameworkAdapter, FrameworkRequest, RequestBasicOptions } from '../framework';
+import { buildAlertsHistogramQuery } from './query.dsl';
+
+import { AlertsAdapter, AlertsGroupData, AlertsBucket } from './types';
+import { TermAggregation } from '../types';
+import { EventHit } from '../events/types';
+
+export class ElasticsearchAlertsAdapter implements AlertsAdapter {
+  constructor(private readonly framework: FrameworkAdapter) {}
+
+  public async getAlertsHistogramData(
+    request: FrameworkRequest,
+    options: RequestBasicOptions
+  ): Promise<AlertsOverTimeData> {
+    const dsl = buildAlertsHistogramQuery(options);
+    const response = await this.framework.callWithRequest<EventHit, TermAggregation>(
+      request,
+      'search',
+      dsl
+    );
+    const totalCount = getOr(0, 'hits.total.value', response);
```

Comment:
nit - I think that we can use the latest of typescript `response?.hits.?total?.value ?? 0`

Reply:
I tried but haven't changed this line as somehow my editor is not happy with it...

Reply:
I think that you have the same problem than me and andrewG told me to install that https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next